src/ui/Button/Button.tsx

```diff
@@ -95,12 +94,46 @@ const loadingVariantClasses = {
   secondary: `disabled:text-white disabled:border-ds-pink-tertiary disabled:bg-ds-pink-default`,
 }
 
-function pickVariant(variant, loading) {
-  const set = loading ? loadingVariantClasses : variantClasses
+function pickVariant(
+  variant: keyof typeof variantClasses | keyof typeof loadingVariantClasses,
+  loading: boolean
+) {
+  return loading
+    ? loadingVariantClasses[variant as keyof typeof loadingVariantClasses]
+    : variantClasses[variant as keyof typeof variantClasses]
+}
 
-  return set[variant]
+// using this type until AppLink is converted to TypeScript
```

Comment:
You maybe able to get the TS types like so: `type Test = PropTypes.InferProps<typeof User.propTypes>`